hypothesis-python/src/hypothesis/strategies/_internal/collections.py

```diff
@@ -316,18 +342,19 @@ class FixedDictStrategy(SearchStrategy):
 
     def __init__(
         self,
-        mapping: dict[T, SearchStrategy[Ex]],
+        mapping: dict[Any, SearchStrategy[Any]],
         *,
-        optional: Optional[dict[T, SearchStrategy[Ex]]],
+        optional: Optional[dict[Any, SearchStrategy[Any]]],
```

Comment:
What's the problem that requires `Any` here?

Reply:
T / Ex works fine but I thought `Any` was more clear, as `FixedDictStrategy` is not actually (yet) parameterized by T and Ex and so those typevars fall back to `Any` anyway.

Reply:
> Hmm, can we parametrize the class with non-Any?

Sure! `{"a": st.integers()}` ideally produces `FixedDictStrategy[str, int]`, though getting that level of precision on the outer `def fixed_dictionaries` itself may be difficult.

> non-dict types

This typing inconsistency comes up frequently for us :( I'll leave a note, but not following up now